types/block_test.go

```diff
@@ -906,3 +906,139 @@ func TestMessagesIsSorted(t *testing.T) {
 		})
 	}
 }
+
+func TestDataToProto(t *testing.T) {
+	type testCase struct {
+		name  string
+		input *Data
+		want  tmproto.Data
+	}
+	testCases := []testCase{
+		{
+			name:  "empty data",
+			input: &Data{Txs: []Tx{}, Evidence: EvidenceData{}, Blobs: []Blob{}, SquareSize: 0},
+			want: tmproto.Data{
+				Txs:        [][]uint8(nil),
+				Evidence:   tmproto.EvidenceList{Evidence: []tmproto.Evidence{}},
+				Blobs:      []tmproto.Blob{},
+				SquareSize: 0x0,
+				Hash:       []uint8(nil),
+			},
+		},
+		{
+			name: "one blob",
+			input: &Data{
+				Txs:      []Tx{},
+				Evidence: EvidenceData{},
+				Blobs: []Blob{
+					{
+						NamespaceID:  []byte{1, 2, 3, 4, 5, 6, 7, 8},
+						Data:         []byte{1},
+						ShareVersion: 0,
+					},
+				},
+				SquareSize: 0,
+			},
+			want: tmproto.Data{
+				Txs:      [][]uint8(nil),
+				Evidence: tmproto.EvidenceList{Evidence: []tmproto.Evidence{}},
+				Blobs: []tmproto.Blob{
+					{
+						NamespaceId:  []uint8{1, 2, 3, 4, 5, 6, 7, 8},
+						Data:         []uint8{1},
+						ShareVersion: 0x0,
+					},
+				},
+				SquareSize: 0x0, Hash: []uint8(nil),
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.input.ToProto()
+			assert.Equal(t, tc.want, got)
+		})
+	}
+}
+
+func TestDataFromProto(t *testing.T) {
+	type testCase struct {
+		name    string
+		input   *tmproto.Data
+		want    Data
+		wantErr bool
+	}
+	testCases := []testCase{
+		{
+			name: "empty data",
+			input: &tmproto.Data{
```

Comment:
it might make sense to refactor these test cases since both tests use the same ones. Then it would be easier to extend the test cases for both tests.

Reply:
resolved in https://github.com/celestiaorg/celestia-core/pull/890/commits/1521fe28560a362b1be8422e1880da80100f5414